src/btree/bt_sync_obsolete.c

```diff
@@ -35,48 +44,56 @@ __sync_obsolete_inmem_evict(WT_SESSION_IMPL *session, WT_REF *ref)
         return (0);
 
     /*
-     * Initialize the time aggregate via the merge initialization, so that stop visibility is copied
-     * across correctly. That is we need the stop timestamp/transaction IDs to start as none,
-     * otherwise we'd never mark anything as obsolete.
+     * There are two places to look for obsolete content, either a saved reconciliation structure or
+     * a reference to an on-disk page via the address structure. If there is a saved reconciliation
+     * result associated with this page use it - that was created more recently, and it's creation
+     * caused the image associated with the address to become obsolete already.
      */
-    WT_TIME_AGGREGATE_INIT_MERGE(&newest_ta);
-    do_visibility_check = obsolete = ovfl_items = false;
-
     mod = ref->page->modify;
-    if (mod != NULL && mod->rec_result == WT_PM_REC_EMPTY) {
-        tag = "reconciled empty";
-
-        obsolete = true;
-    } else if (mod != NULL && mod->rec_result == WT_PM_REC_MULTIBLOCK) {
-        tag = "reconciled multi-block";
-
-        /* Calculate the max stop time point by traversing all multi addresses. */
-        for (multi = mod->mod_multi, i = 0; i < mod->mod_multi_entries; ++multi, ++i) {
-            WT_TIME_AGGREGATE_MERGE_OBSOLETE_VISIBLE(session, &newest_ta, &multi->addr.ta);
-            if (multi->addr.type == WT_ADDR_LEAF)
+    if (mod != NULL && mod->rec_result != 0) {
```

Comment:
Thanks @agorrod. I agree it's not making it cleaner here. We can use either solution. In the second case, we only have `ref->state` in the if check, refactoring it to switch should be easier?